### PR TITLE
Fix Attribute Serialization

### PR DESCRIPTION
**Pull request by KevinRansom**
Fixes: [#15140](https://github.com/dotnet/fsharp/issues/15140)

F# Attribute blob serialization for open generics created an invalid blob.  This addresses that issue.

The fix is to use typdefof, which serializes correctly for open generics.

Todo:

- [x] Add regression

### CODE REVIEW

File: tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/CustomAttributes/Basic/EnsureValidCustomAttributeBlob.fs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+open System
+open System.Reflection
+open FSharp.Collections
+
+let attributes = typeof<List<int>>.GetConstructor(BindingFlags.NonPublic ||| BindingFlags.Instance, Unchecked.defaultof<Binder>, Type.EmptyTypes, [||]).GetCustomAttributes()
+for a in attributes do Console.WriteLine(a.ToString())
```
Comment (T-Gro) [inline tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/CustomAttributes/Basic/EnsureValidCustomAttributeBlob.fs:6]: This is where it used to fail I assume?

File: tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj [hunk 2]
```diff
@@ -263,6 +263,7 @@
   <ItemGroup>
     <Content Include="resources\**" CopyToOutputDirectory="Never" CopyToPublishDirectory="PreserveNewest" />
     <EmbeddedResource Remove="Properties\**" />
+    <None Remove="EmittedIL\GenericComparison\Hash09.fsx.il.bsl" />
```
Comment (T-Gro) [inline tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj:266]: Do we need .bsl within .fsproj?

**Review by T-Gro — APPROVED** [review]

**Review by abonie — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/CustomAttributes/Basic/EnsureValidCustomAttributeBlob.fs (hunk 1 above)
Comment (KevinRansom) [reply tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/CustomAttributes/Basic/EnsureValidCustomAttributeBlob.fs:6]: it is the repro yes.

File: tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj (hunk 2 above)
Comment (KevinRansom) [reply tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj:266]: The baselines are displayed in the ui, even though they aren't added persisted.  VS adds this remove, when you do a delete.  I will delete the remove this line..

**Pull request merged**